app/models/ems_folder.rb

```diff
@@ -166,6 +166,26 @@ def folder_path_objs(*args)
     folders
   end
 
+  # similar to folder_path_objs, but called on a scope
+  def self.folder_path_objs(relationship_type, *args)
+    options = args.extract_options!
+    folder_rels = Relationship.find_by(:resource_id   => select(:id),
+                                       :resource_type => "EmsFolder",
+                                       :relationship  => relationship_type)
+    if folder_rels.nil? # parent folder has no parents, use the parent folder
+      folders = [first].compact
+    else
+      folder_path_resource_ids = folder_rels.path.where(:resource_type => "EmsFolder",
+                                                        :relationship  => relationship_type)
+                                                 .select(:resource_id)
+      folders = EmsFolder.where(:id => folder_path_resource_ids)
+    end
+    # we can translate these into scopes if needed
+    folders = folders[1..-1] if options[:exclude_root_folder]
+    folders = folders.reject(&:hidden?) if options[:exclude_non_display_folders]
```

Comment:
Since both of the methods have this part the same, can we move the common code into a private method?